controller/lib/hal/hal.h

```diff
@@ -146,11 +147,16 @@ class HalApi {
   // analogRead() reads the value of an analog input pin. analogWrite() writes
   // to a PWM pin - some of the digital pins are PWM pins.
 
+  // Reads from analog sensor.
+  //
+  // On Arduino, the analog-to-digital converter has 10 bits of precision, so
+  // you get a number in the range [0, 1024).
```

Comment:
Is this still going to be the case when we're not on Arduino?

Reply:
Nope, this calculation needs to be HAL-ified.

Reply:
Presumably when we hal-ify it, we'll return a float in the range [0-1]?  But also hope it's OK to leave this be for now.

Reply:
Just be warned...if someone tries to use any of these related methods on STM32 the results will be quite wrong. Simple difference of VREF_P/4096 instead. (I'm not sure whether the Nucleo uses 3.3 or 5 V positive reference off the top of my head).

Reply:
Yikes, thanks.  cc @Miceuz, who's been leading the porting effort.  I'll also add a TODO.